### PR TITLE
projects: Add "Improve Diff Handling"

### PR DESCRIPTION
**Pull request by sils**

### CODE REVIEW

**Comment by hemangsk** [comment]
ack a035fd0

File: data/projects.js [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
-		"mentors" : [
-		"user1",
-		"user2"
+		"name" : "Improve Diff Handling",
+		"desc" : "Bears should be able to offer more than just one possible patch \
+for an issue. Users should be able to choose between different patches, e.g. \
+for different possible spelling corrections.\
+\
+In addition bears should be able to offer templated patches, e.g. to offer \
+variable renamings while the user should be asked for the new variable name.",
```
Comment (sims1253) [inline data/projects.js]: With all these linebreaks the code is not nice to read. Any better formatting options?

Comment (sims1253) [inline data/projects.js]: Also maybe we want to give simple examples?

File: data/projects.js [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
+		"requirements" : [
+			"good concept in the proposal",
```
Comment (sims1253) [inline data/projects.js]: isn't this a general requirement?

File: data/projects.js [hunk 3]
```diff
@@ -1,10 +1,38 @@
...
+			"familiarity with the coala core"
```
Comment (sims1253) [inline data/projects.js]: Add python?

File: data/projects.js [hunk 4]
```diff
@@ -1,10 +1,38 @@
...
+				"A cEP for the project has been merged."
+			],
+			"GSoC 2017 Midterm" : [
+				"The user is able to choose between multiple patches.",
+				"At least one bear exists that makes use of this."
+			],
+			"GSoC 2017 Final" : [
+				"The user is able to apply templated patches.",
+				"At least one bear exists that makes use of this."
+			]
```
Comment (sims1253) [inline data/projects.js]: Add full documentation to milestones.

Comment (sils) [inline data/projects.js]: what do you mean with full documentation?

File: data/projects.js (hunk 2 above)
Comment (sils) [inline data/projects.js]: well for some it's more important I think

File: data/projects.js (hunk 1 above)
Comment (sils) [inline data/projects.js]: not really better formatting options :/

File: data/projects.js (hunk 4 above)
Comment (sims1253) [inline data/projects.js]: they should write the documentation for what they add themselves.

File: data/projects.js (hunk 1 above)
Comment (sils) [inline data/projects.js]: isn't variable renaming a good example?

Comment (sims1253) [inline data/projects.js]: i meant something like a mockup

**Pull request merged**